### PR TITLE
Updates to `PaymentRequest` stuctures to align with the latest TSC spec.

### PR DESCRIPTION
**Pull request by AustEcon**
As discussed with @jadwahab and @sirdeggen here is the PR for updating to the latest TSC spec: https://tsc.bitcoinassociation.net/standards/direct_payment_protocol/

Corresponding dual PR here: https://github.com/bitcoin-sv/dpp-proxy/pull/57

So far this only updates the `PaymentRequest` structure which is enough for now to get ElectrumSV through the first part of the data flow for development...

Still to do:

- [x] `Payment`
- [x] `PaymentACK`
- [x] `Adequate testing end-to-end against this code`

Until the `dpp-proxy` is updated to the TSC spec, ElectrumSV will be building against this feature branch to avoid technical debt of the old message structures. So if you are not a fan of this implementation, that is no problem - it still unblocks Roger and I to keep progressing the ElectrumSV integration until the dpp-proxy master branch catches up. 😃

### CODE REVIEW

File: payment_request.go [hunk 1]
```diff
@@ -2,45 +2,93 @@ package dpp
...
+
+// PaymentModes message used in DPP TSC spec.
+// At present we will strictly only allow the "standard" mode of payment with native bitcoins (satoshis). Handling
+// of tokens is left for a later date.
+type PaymentModes struct {
+	HybridPaymentMode map[string]map[string][]TransactionTerms `json:"ef63d9775da5"`
+}
+
+// PaymentRequest message as defined in the DPP T$C spec.
 type PaymentRequest struct {
```
Comment (jadwahab) [inline payment_request.go]: also shouldn't this be renamed to `PaymentTerms`? 🤔

Comment (AustEcon) [reply payment_request.go]: Yes. You are right. I will change it

File: payment_request.go [hunk 2]
```diff
@@ -57,5 +104,5 @@ type PaymentRequestService interface {
 
 // PaymentRequestReader will return a new payment request.
 type PaymentRequestReader interface {
```
Comment (jadwahab) [inline payment_request.go]: shouldn't all places where we have PaymentRequest be PaymentTerms instead?

**Review by jadwahab — CHANGES_REQUESTED** [review]
I think we should have different folders/modules for different payment modes, what do you think? would be worth asking @theflyingcodr

File: payment_request.go (hunk 2 above)
Comment (AustEcon) [reply payment_request.go]: Yes. I suppose it should. I have now updated all instances of `PaymentRequest` to `PaymentTerms` for go-dpp and the dpp-proxy.

This includes the websocket message types: `paymentterms.create`, `paymentterms.response` and `paymentterms.error`

For consistency.

**Comment by AustEcon** [comment]
> I think we should have different folders/modules for different payment modes, what do you think? would be worth asking @theflyingcodr

Seems like a good suggestion.

I have refactored into a `modes` package with nested `hybridmode`. LMK if that's kind of what you're thinking.

**Comment by sirdeggen** [comment]
Still discussing, sorry @AustEcon - early next week should have resolved the differences in opinion on this.

**Pull request closed without merging**